vendor/github.com/ethereum/go-ethereum/whisper/whisperv5/whisper.go

```diff
@@ -718,6 +735,44 @@ func (wh *Whisper) add(envelope *Envelope) (bool, error) {
 	return true, nil
 }
 
+func (w *Whisper) sendIncomingMessageState(isP2P bool, status message.Status, src NewMessage, env *Envelope, rec *ReceivedMessage, err error) {
```

Comment:
I think both functions differs only by "incoming/outgoing" parameters, so they can be merged into one even further :)

Also, what do you think about renaming it to just "trace()"? As it technically a tracing events feature. "Sending message state through deliver server" sounds a bit cryptic for someone who might see this code for the first time.

Reply:
Hmmmm, i chose the two method approach to allow clean seperation and also reduce function arguments, we definitely can merge both, but honestly i feel more that we should keep them both, it feels more sensible at first glance what is being sent out.

On the other hand, i do agree that using `traceDeliveryEvent`, `traceIncomingDeliveryEvent` is a far more better name.